qemu/tests/blockdev_snapshot_merge.py

```diff
@@ -0,0 +1,83 @@
+import logging
+
+from provider.blockdev_snapshot_base import BlockDevSnapshotTest
+
+
+class BlockdevSnapshotMergeTest(BlockDevSnapshotTest):
+    def pre_test(self):
+        if not self.main_vm.is_alive():
+            self.main_vm.create()
+        self.main_vm.verify_alive()
+        if self.base_tag != "image1":
+            self.configure_data_disk()
+
+    def prepare_snapshot_file(self):
+        self.params["image_size_%s" % self.snapshot_tag] = self.base_image.size
+        self.params["image_name_%s" % self.snapshot_tag] = 'images/' + self.snapshot_tag
+        snapshot_format = self.params.get("snapshot_format", "qcow2")
+        self.params["image_format_%s" % self.snapshot_tag] = snapshot_format
+        super(BlockdevSnapshotMergeTest, self).prepare_snapshot_file()
+
+    def verify_snapshot(self):
+        if self.main_vm.is_alive():
+            self.main_vm.destroy(free_mac_addresses=False)
+        self.snapshot_image_list = []
+        snapshot = self.snapshot_tags[-1]
+        for item in self.snapshot_tags[-2::-1]:
+            base_image = self.get_image_by_tag(item)
+            snapshot_image = self.get_image_by_tag(snapshot)
+            self.snapshot_image_list.append(snapshot_image)
+            snapshot_image.base_tag = item
+            snapshot_image.base_format = base_image.get_format()
+            base_image_filename = base_image.image_filename
+            snapshot_image.base_image_filename = base_image_filename
+            snapshot_image.rebase(snapshot_image.params)
+            snapshot_image.commit()
+            snapshot = item
+        self.clone_vm = self.main_vm
+        self.clone_vm.create()
+        self.clone_vm.verify_alive()
+        if self.base_tag != "image1":
+            self.mount_data_disks()
+        self.verify_data_file()
+
+    def snapshot_test(self):
+        snapshot_nums = self.params.get_numeric("snapshot_num")
+        self.snapshot_tags = [self.base_tag]
+        for index in range(1, snapshot_nums + 1):
+            self.snapshot_tag = "sn%s" % index
+            if self.snapshot_tag not in self.snapshot_tags:
+                self.snapshot_tags.append(self.snapshot_tag)
+            self.prepare_snapshot_file()
```

Comment:
@aliang123 All the snapshot files used the same name /mnt/vdb1/data: 

22:31:50 DEBUG| Sending command: dd if=/dev/urandom of=/mnt/vdb1/data bs=1M count=10 oflag=direct
22:31:50 DEBUG| Sending command: echo $?
22:31:50 DEBUG| Sending command: md5sum /mnt/vdb1/data > /mnt/vdb1/data.md5 && sync

Reply:
@ngu-niny, updated it, please help to review again, thanks.